### PR TITLE
Add missing Reader callbacks

### PR DESCRIPTION
**Pull request by EricLin-BBpos**
## Summary

Add missing callbacks on ReaderListener and BluetoothReaderDelegate
## Motivation

Ensure all the callbacks on native SDKs are supported:

## Testing

- [x] I tested this manually
- [ ] I added automated tests

## Documentation

Select one:

- [ ] I have added relevant documentation for my changes.
- [x] This PR does not result in any developer-facing changes.

### CODE REVIEW

File: android/src/main/java/com/stripeterminalreactnative/ReactNativeConstants.kt [hunk 1]
```diff
@@ -19,5 +19,8 @@ enum class ReactNativeConstants(val listenerName: String) {
     CHANGE_OFFLINE_STATUS("didChangeOfflineStatus"),
     FORWARD_PAYMENT_INTENT("didForwardPaymentIntent"),
     REPORT_FORWARDING_ERROR("didReportForwardingError"),
-    DISCONNECT("didDisconnect")
+    DISCONNECT("didDisconnect"),
+    BATTERY_LEVEL_UPDATE("didBatteryLevelUpdate"),
```
Comment (nazli-stripe) [inline android/src/main/java/com/stripeterminalreactnative/ReactNativeConstants.kt:23]: `didUpdateBatteryLevel` ?

File: ios/Mappers.swift [hunk 2]
```diff
@@ -833,6 +833,14 @@ class Mappers {
 
         return (["collectInputResults": collectInputResults])
     }
+
+    class func mapFromReaderEvent(_ readerEvent: ReaderEvent) -> String {
+        switch readerEvent {
+            case ReaderEvent.cardInserted: return "cardInserted"
+            case ReaderEvent.cardRemoved: return "cardRemoved"
+            default: return "unknown"
+        }
```
Comment (bric-stripe) [inline ios/Mappers.swift:842]: recommend using `@unknown default` so the compiler tells us when new entries are added

and should be able to remove the prefix

```suggestion
        switch readerEvent {
            case .cardInserted: return "cardInserted"
            case .cardRemoved: return "cardRemoved"
            @unknown default: return "unknown"
        }
```

Comment (bric-stripe) [reply ios/Mappers.swift:842]: and should the values be the values from the enum?

```
export enum ReaderEvent {
  CARD_INSERTED = 'CARD_INSERTED',
  CARD_REMOVED = 'CARD_REMOVED',
}
```

? (not sure what's expected on the RN layer there - or does something else map them to that?)

**Review by ugochukwu-stripe — COMMENTED** [review]
Android changes lgtm

File: ios/StripeTerminalReactNative.swift [hunk 3]
```diff
@@ -21,6 +21,9 @@ enum ReactNativeConstants: String, CaseIterable {
     case FORWARD_PAYMENT_INTENT = "didForwardPaymentIntent"
     case REPORT_FORWARDING_ERROR = "didReportForwardingError"
     case DISCONNECT = "didDisconnect"
+    case UPDATE_BATTERY_LEVEL = "didReportBatteryLevel"
```
Comment (nazli-stripe) [inline ios/StripeTerminalReactNative.swift:24]: `didUpdateBatteryLevel`

**Review by nazli-stripe — DISMISSED** [review]

**Review by nazli-stripe — APPROVED** [review]

**Pull request merged**